Google.Api.Generator/Generation/ServiceCodeGenerator.cs

```diff
@@ -78,6 +82,37 @@ private static IEnumerable<MemberDeclarationSyntax> PaginatedPartialClasses(Sour
             }
         }
 
+        private static MemberDeclarationSyntax PaginatedPartialInterfaceClass(SourceFileContext ctx, Typ typ, MessageDescriptor messageDesc)
+        {
+            var partialInterfaceCls = Class(Public | Partial, typ, baseTypes: ctx.Type<IPageRequest>());
+
+            if (messageDesc.FindFieldByName("page_size") is null)
+            {
+                //DiREGapic scenario where `max_results` is an option for a page size-semantic field.
+                var maxResMessage = messageDesc.FindFieldByName("max_results");
+                if (maxResMessage is null)
+                {
+                    throw new InvalidOperationException("Paginated request should have either page_size or max_results field.");
+                }
+
+                using (ctx.InClass(partialInterfaceCls))
+                {
+                    var underlyingProperty = Property(DontCare, ctx.TypeDontCare, "MaxResults");
+
+                    var assignmentVal = ProtoTyp.Of(maxResMessage) == Typ.Of<int>() 
+                        ? Value
+                        : CheckedCast(ctx.Type(ProtoTyp.Of(maxResMessage)), Value);
+
+                    var property =  Property(Public, ctx.Type<Int32>(), "PageSize")
```

Comment:
Have you considered making this an explicitly-implemented property instead of a public one? I think that might be less confusing, if it works.

Reply:
Turns out this does not quite work because we do generate a code accessing the public setter, e.g.:
```
        public virtual gax::PagedEnumerable<AddressAggregatedList, scg::KeyValuePair<string, AddressesScopedList>> AggregatedList(string project, string pageToken = null, int? pageSize = null, gaxgrpc::CallSettings callSettings = null) =>
            AggregatedList(new AggregatedListAddressesRequest
            {
                Project = gax::GaxPreconditions.CheckNotNullOrEmpty(project, nameof(project)),
                PageToken = pageToken ?? "",
                PageSize = pageSize ?? 0,
            }, callSettings);
```

I have followed your suggestions in the Compute PR and have added code for generating a public getter alongside the setter.

Reply:
Ah, that's a pity. (And now you mention it, that makes sense as I think I tried to do it myself.)

We could *potentially* set MaxResults instead, but making PageSize public instead is okay.